src/client/Client.cc

```diff
@@ -11520,15 +11534,20 @@ int64_t Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
     ceph_assert(in->inline_version > 0);
   }
 
-  // copy into fresh buffer (since our write may be resub, async)
+  // make a bufferlist, if not zerocopy make a copy
   bufferlist bl;
   if (buf) {
+    assert(!zerocopy);
     if (size > 0)
       bl.append(buf, size);
   } else if (iov){
     for (int i = 0; i < iovcnt; i++) {
       if (iov[i].iov_len > 0) {
-        bl.append((const char *)iov[i].iov_base, iov[i].iov_len);
+        if (zerocopy)
+          bl.push_back(buffer::create_static(iov[i].iov_len,
+                                             static_cast<char*>(iov[i].iov_base)));
+        else
+          bl.append((const char *)iov[i].iov_base, iov[i].iov_len);
```

Comment:
```suggestion
          bl.append(static_cast<const char*>(iov[i].iov_base), iov[i].iov_len);
```
better to use C++ style here (since we're using the same in `if {...}` too)

Reply:
sure